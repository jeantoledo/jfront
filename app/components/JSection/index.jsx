import React from 'react';
import PropTypes from 'prop-types';

const JSection = props => (
  <section className="section">
    <div className="container is-fluid">
      <div className="columns">
        { props.logo ?
          <div className="column is-narrow">
            <img src={props.logo} alt="" />
          </div>
        : null }
        <div className="column">
          <h1 className="title">{props.title}</h1>
          <h2 className="subtitle">{props.subtitle}</h2>
        </div>
      </div>
    </div>
  </section>
);

JSection.defaultProps = {
  logo: '',
};


JSection.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  logo: PropTypes.string,
};

export default JSection;
