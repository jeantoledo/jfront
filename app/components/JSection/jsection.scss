.jtitle {
  display: inline-block;
  padding: 2px;
  border: 1px solid #2980b9 !important;
}
