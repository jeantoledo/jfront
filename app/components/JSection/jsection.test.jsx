import React from 'react';
import renderer from 'react-test-renderer';
import JSection from './jsection';

test('JSection render title and subtitle props', () => {
  const component = <JSection title="JFront" subtitle="Javascript Single Page Application" />;
  expect(renderer.create(component)).toMatchSnapshot();
});
