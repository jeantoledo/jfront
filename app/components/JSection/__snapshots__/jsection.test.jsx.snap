// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSection render title and subtitle props 1`] = `
<section
  className="section"
>
  <div
    className="container is-fluid"
  >
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <h1
          className="title jtitle"
        >
          JFront
        </h1>
        <h2
          className="subtitle"
        >
          Javascript Single Page Application
        </h2>
      </div>
    </div>
  </div>
</section>
`;
